Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -677,6 +677,16 @@ ExceptionOr<void> ContainerNode::removeChild(Node& oldChild)
     rebuildSVGExtensionsElementsIfNecessary();
     dispatchSubtreeModifiedEvent();
 
+    if (is<Element>(oldChild)) {
+        Element& element = downcast<Element>(oldChild);
+        if (element.lastRememberedSize()) {
+            // The disconnected element could be unobserved because of other properties, here we need to make sure it is observed,
+            // so that deliver could be triggered and it would clear lastRememberedSize.
+            document().observeForContainIntrinsicSize(element);
+            document().resetObservationSizeForContainIntrinsicSize(element);
+        }
+    }
```

Comment:
You can use `dynamicDowncast`:

```suggestion
    auto* element = dynamicDowncast<Element>(oldChild);
    if (element && element->lastRememberedSize()) {
        // The disconnected element could be unobserved because of other properties, here we need to make sure it is observed,
        // so that deliver could be triggered and it would clear lastRememberedSize.
        document().observeForContainIntrinsicSize(*element);
        document().resetObservationSizeForContainIntrinsicSize(*element);
    }
```

Reply:
Done, thanks!